zap/src/main/java/org/zaproxy/zap/extension/ascan/FilterPanel.java

```diff
@@ -0,0 +1,502 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascan;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Vector;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.swing.DefaultListModel;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.ListSelectionModel;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.SiteNode;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.extension.ascan.filters.GenericFilterUtility;
+import org.zaproxy.zap.extension.ascan.filters.ScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.HttpStatusCodeScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.MethodScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.TagScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.UrlPatternScanFilter;
+import org.zaproxy.zap.model.Target;
+import org.zaproxy.zap.utils.ZapLabel;
+import org.zaproxy.zap.view.LayoutHelper;
+
+/**
+ * FilterPanel is used as a Tab in {@link CustomScanDialog}. This tab gives various options for
+ * filtering the Active Scan messages based on the selection.
+ *
+ * @author KSASAN preetkaran20@gmail.com
+ * @since TODO add version
+ */
+class FilterPanel extends JPanel {
+
+    private static final long serialVersionUID = 1L;
+
+    private JPanel jPanel1 = null;
+
+    private JButton btnReset = null;
+    private JPanel jPanel2 = null;
+
+    private JList<String> methodList = null;
+    private JList<Integer> codeList = null;
+    private JList<String> incTagList = null;
+    private JList<String> excTagList = null;
+    private JTextArea regexInc = null;
+    private JTextArea regexExc = null;
+
+    private DefaultListModel<String> tagModel = null;
+
+    private JScrollPane methodScroller = null;
+    private JScrollPane codeScroller = null;
+    private JScrollPane incTagScroller = null;
+    private JScrollPane excTagScroller = null;
+
+    private JScrollPane urlRegxIncScroller = null;
+    private JScrollPane urlRegxExcScroller = null;
+
+    // Constructs and Initializes the Panel
+    public FilterPanel(Target target) {
```

Comment:
I'd suggest removing the parameter, not really needed.

Reply:
sure.